packages/typescript-estree/src/parser.ts

```diff
@@ -57,8 +59,43 @@ function getProgramAndAST(
     }
   }
 
+  // todo; wrap in function or some such
+  if (parseSettings.projectService) {
+    parseSettings.projectService.openClientFile(
+      parseSettings.filePath,
+      parseSettings.codeFullText,
+    );
+
+    // todo: maybe we'll slap this into the parseSEttings as a property?
+    // const projectService = createProjectServiceProgram(/* parseSettings */);
+    const program = parseSettings.projectService
+      .getScriptInfo(parseSettings.filePath)!
+      .getDefaultProject()
+      .getLanguageService(/*ensureSynchronized*/ true)
+      .getProgram();
+
+    // TODO "Eventually, close the file"
+    // what does this actually do?
+    //
+
+    if (program) {
+      const projectServiceProgram = createProjectProgram(parseSettings, [
+        // LOL TYPES
+        // This'll hoepfully be resolved once they fully move to modules
+        // right now it's two separate files that get dumped out
+        program as ts.Program,
```

Comment:
> `tsserverlibrary` is an entirely different module than `typescript`.

Yeah, `ts.Program is not assignable to ts.Program` here. Perhaps this change should be blocked on TypeScript fully moving to ESM and having one version of each class/enum/etc.?

Reply:
The problem is that you basically need to be using tsserverlibrary the whole time, because it really is a differnet bundle.

As much as I want ESM to be "the thing", I don't know if it helps you all given you can't have ESM for eslint and you can't use async either?

Reply:
Hmm, that might take a while: https://github.com/eslint/eslint/issues/15475